rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -282,6 +282,152 @@ impl<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>> ArrowBlockfileRe
         }
     }
 
+    pub(crate) async fn get_gt(
+        &'me self,
+        prefix: &str,
+        key: K,
+    ) -> Result<Vec<(&str, K, V)>, Box<dyn ChromaError>> {
+        // Get all block ids that contain keys > key from sparse index.
+        let search_key = CompositeKey::new(prefix.to_string(), key.clone());
+        let block_ids = self.sparse_index.get_block_ids_gt(search_key);
+        let mut result: Vec<(&str, K, V)> = vec![];
+        // Read all the blocks individually to get keys > key.
+        for block_id in block_ids {
+            let block_opt = self.get_block(block_id).await;
+            let block = match block_opt {
+                Some(b) => b,
+                None => {
+                    return Err(Box::new(ArrowBlockfileError::BlockNotFound));
+                }
+            };
+            match block.get_gt(prefix, key.clone()) {
+                Some(data) => {
+                    result.extend(data);
+                }
+                None => {
+                    return Err(Box::new(BlockfileError::NotFoundError));
+                }
+            };
+        }
+        return Ok(result);
+    }
+
+    pub(crate) async fn get_lt(
+        &'me self,
+        prefix: &str,
+        key: K,
+    ) -> Result<Vec<(&str, K, V)>, Box<dyn ChromaError>> {
+        // Get all block ids that contain keys < key from sparse index.
+        let search_key = CompositeKey::new(prefix.to_string(), key.clone());
+        let block_ids = self.sparse_index.get_block_ids_lt(search_key);
+        let mut result: Vec<(&str, K, V)> = vec![];
+        // Read all the blocks individually to get keys < key.
+        for block_id in block_ids {
+            let block_opt = self.get_block(block_id).await;
+            let block = match block_opt {
+                Some(b) => b,
+                None => {
+                    return Err(Box::new(ArrowBlockfileError::BlockNotFound));
+                }
+            };
+            match block.get_lt(prefix, key.clone()) {
+                Some(data) => {
+                    result.extend(data);
+                }
+                None => {
+                    return Err(Box::new(BlockfileError::NotFoundError));
+                }
+            };
+        }
+        return Ok(result);
+    }
+
+    pub(crate) async fn get_gte(
+        &'me self,
+        prefix: &str,
+        key: K,
+    ) -> Result<Vec<(&str, K, V)>, Box<dyn ChromaError>> {
+        // Get all block ids that contain keys >= key from sparse index.
```

Comment:
You can use a triple /// and put this above the function for better use of rustdocs